### ISSUE TITLE
BUG: KernelExplainer really slow on sklearn SVM with RBF Kernel

### ISSUE THREAD

**Issue by sultanoid** [shap/shap#3747]
### Issue Description

I am trying to use the KernelExplainer on a SVM binary classifier as follows:

`explainer = shap.KernelExplainer(svm_model.predict_proba, X_train_shap_numpy[:100])`
       
However, when I try to extract the shap values using

` shap_values = explainer.shap_values(test_data_numpy)  `

it seems to take a prohibitive amount of time.

` 0%|          | 3/200 [08:50<10:57:27, 176.78s/it]`

I am trying to use SHAP library to explain important features globally. 

From this reply, I have come to know that,  for the KernelExplainer one should put a warning message in when using large background samples. 200 rows means the kernel explainer will be 200 times slower than for 1 sample.

[https://github.com/shap/shap/issues/66#issuecomment-384674879](https://github.com/shap/shap/issues/66#issuecomment-384674879)

But then, what explainer of SHAP libary should I use to idenitfy important features globally for at least 200 samples? 

### Minimal Reproducible Example

```python
import shap 

explainer = shap.KernelExplainer(svm_model.predict_proba, X_train_shap_numpy[:100])
shap_values = explainer.shap_values(test_data_numpy)
```

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.44.1

**Comment by btsai-dev** [comment]
Adding onto this thread, it is likely that the problem lies with scikit-learn's SVM implementation, mentioned in https://github.com/scikit-learn/scikit-learn/issues/21410

To fix this, you use the `rbf_kernel` defined in `sklearn.metrics.pairwise` which takes advantage of numpy's SIMD and memory optimizations, instead of just passing in the `"rbf"` string. Furthermore the resulting performance is different between the two. You will also incur a higher memory cost from using the `rbf_kernel` function.
